### PR TITLE
fix(sled): Don't derive a StoreCipher twice if we're reusing the db

### LINKED ISSUE #594: storecipher for passphrase derivition

### PR DESCRIPTION
**Pull request by poljar**
This closes #594.

### CODE REVIEW

File: crates/matrix-sdk-sled/src/cryptostore.rs [hunk 1]
```diff
@@ -347,7 +357,7 @@ impl SledStore {
     fn open_helper(
         db: Db,
         path: Option<PathBuf>,
-        passphrase: Option<&str>,
+        store_cipher: Arc<Option<StoreCipher>>,
```
Comment (jplatte) [inline crates/matrix-sdk-sled/src/cryptostore.rs:360]: `Arc<Option<_>>` seems like a weird combo. Would `Option<Arc<_>>` not work?

**Review by jplatte — APPROVED** [review]
One nit but it's really about pre-existing things. The actual change looks good 👍🏼

**Review by jplatte — APPROVED** [review]

**Comment by poljar** [comment]
Actually the indexeddb does this as well, so there's now one more commit.

**Review by jplatte — APPROVED** [review]
Another approval it is! 😄

**Pull request merged**